### DISCUSSION TITLE
Getting Started (1) | Add HACS & SVG-files and YAML entity mapping in Home Assistant

### DISCUSSION THREAD

**Discussion by exetico** [ExperienceLovelace/ha-floorplan#249] [Videos]
### Getting Started video for ha-floorplan

Hi. Today is ... a new day for me.

Yesterday i planned and recorded a video on how to get started with ha-floorplan. It's done with Home Assistant OS, which are the easiest way to get started with Home Assistant.

[![image](https://user-images.githubusercontent.com/3549445/196247820-1954010a-5c52-4a58-b572-4edde9040575.png)](https://youtu.be/5ue9J4WG2a8)

https://youtu.be/5ue9J4WG2a8

I really hope that I'll find time to create more videos in the future, with more specific content - but I've received quite a few requests on how to get started with ha-floorplan, in the last couple of years.

_It's my first video in English and as a "teacher", but I've executed it the best I can, during a rainy sunday._

Attachments can be found here:
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/tree/main/1_Getting-Started-with-ha-floorplan

**Please hit the like-button on the video, and write a comment.** **Thanks!** 🙏

If you have any questions related to the video, fell free to drop them in this discussion :smiley:.

<hr>

### Links

Share your first frustrations about ha-floorplan here: https://github.com/ExperienceLovelace/ha-floorplan/discussions/247
Buy me a coffee ☕: https://www.buymeacoffee.com/coffeetoexetico

**Comment by OzGav** [comment]
EXCELLENT! Thanks Tobias looks great. Will be interesting to see what others say they would like to have explained. I wonder if they might ask for a simple example of how to create the initial files?

In this video you have used the Home example but you could use a simple one or two room blank plan and show how to add a single light bulb. From there you could explain when you need to use the element and entity lines in the yaml and when you only need entity. You could then show how to change the image of the bulb using image_set as you did in this one but also class_set by showing and hiding a top image and maybe even style_set. 

In this way you will have shown the relationship between the SVG, YAML and CSS which I think is often misunderstood.

But again this was great and shows how to use so many parts of HA not just floorplan! Thanks again!

**Reply by exetico** [reply]
Thanks for the feedback! If I find the time, I'll totally make a few shorter ones, focusing on on element instead of multiple. But my initial throughs was, that most people is still getting stuck in the first few steps, before they actually have a ha-floorplan setup running ☺️.

The next parts should be more easier to jump on, and I'll also try to improve my recording-workflow, so it's easier to post-edit.

**Reply by OzGav** [reply]
Yep absolutely this one is great

**Reply by exetico** [reply]
The scope is set for the next video, maybe separated into multiple smaller ones.

More than three ours spend today, so hopefully I'm able to record it, in one of the next days...

Sneak peak:
![image](https://user-images.githubusercontent.com/3549445/196751911-7c17e982-2b87-46d2-9d0a-38c7baa15a53.png)
![image](https://user-images.githubusercontent.com/3549445/196751073-ffbc2464-7f20-4d96-b6f9-a19ceb9135e9.png)
![image](https://user-images.githubusercontent.com/3549445/196751178-06a7200d-aef9-4406-9ef7-b1c367a00684.png)
![image](https://user-images.githubusercontent.com/3549445/196751236-238b9328-9ca9-4eb5-be44-c76ffbf43ae6.png)

**Reply by OzGav** [reply]
Looks good. I would just suggest in step 3 keeping it simple to start with perhaps a smaller CSS and YAML than what you have shown above. And as I mentioned above it would be good if you can carefully explain the relationship between the SVG, YAML and CSS.  This will be so helpful to everyone.

**Reply by exetico** [reply]
I'll split it into a two part as originally planned. The relation will be part of the first video.

The thing you see, is the "final product". My plan was to create all three files in the two-part video, while explaining the relations.

**Reply by exetico** [reply]
Video 2 part 1 is now on YouTube :-) 

I'll share it in a few other way, when I get time.
https://studio.youtube.com/video/f_3KAPc53hQ/edit

I'm planning to include the leftovers in the part 2 (all things not mentioned and drawn in part 1):
https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/tree/main/2_2_Create-your-own-floorplan-in-Inkscape-and-ha-floorplan-logics

It's recorded with another microphone, but I'm not 100% sure I'm like the most.

**Reply by exetico** [reply]
Discussion here:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/255

**Reply by OzGav** [reply]
Nice. The main thing I think could do with more explanation is when you go to draw the rooms you don't mention why you are doing that. I would say something like:

> The background image I just placed can't be interacted with. So you have to add additonal elements that will be controlled by ha-floorplan. They could be small buttons but in this case I will use a large button (a rectangle) that is actually the size of each room and they will also serve as an indication of the light status as you will eventually see when I talk about selecting the fill colour. Later ha-floorplan will be able to show or hide this rectangle based on the state of the room light.

At 10:35 you say you are going to set a class which is defined in the service_data but I think it might have been also good at that point to say that the classes are contained in the CSS file which you are going to edit shortly. At 10:52 I might have started with a simple example with the class name. You might lose some people here who don't know that you can create the class name dynamically. At 11:21 the sub-titles you have there are great but you intro the next screen by saying "Let's go back" I wonder if it might be better to say something like:

> So we need to add some class definitions to our CSS file now. Let's see how to do that.

And I'm not sure if you can do it with the screen capture software you are using but if you can zoom in on the YAML and CSS that would be great as it is clear but small at the moment.

Lastly, I personally hardly ever use the element key as I assign my SVG elements an ID the same as the entity. I find that simpler to work with so not sure if you want to mention that at some point. Although I do notice that might go against the point you make at the end about naming of the elements??!

The wrap up you do at the end going back over each thing you covered is great. Thanks again for doing these Tobias you are going above and beyond!

**Reply by exetico** [reply]
Maybe you should start creating videos, too. Scoping a 20min video is quite a task 😁.

I'll try and work each steps more into each other. But that task is easier said than done. 

Thank you for the feedback!

This is still primary for people starting the ha-usage for the first time, and going too deep into the technical elements won't always have a good impact on things. So it was a calculated risk, to do it this way. Some people are touching JS, YAML, CSS and HTML for the first time - and It's not possible to fit all knowlegde around that, into each chapters. At least, that's what I think. 

My plan was to dedicate a video to "how ha-floorplan actually works" or simular, for people asking for that kind of details. And also, shorter videos explaining one service at a time, or simular. 

> Lastly, I personally hardly ever use the element key as I assign my SVG elements an ID the same as the entity. I find that simpler to work with so not sure if you want to mention that at some point. Although I do notice that might go against the point you make at the end about naming of the elements??!

I'm not sure that's you're trying to say. That you're just using the ID's generated by InkScape in the first place? I don't think that's the right path to take, for people touching ha-floorplan for the first time.

**Reply by OzGav** [reply]
Yes I coach touch football and I make videos for the girls to watch so I know what you mean. I don’t have a decent screen capture program to be able to do what you are doing. What are you using?

These will be great. New users might have some questions but they won’t be staring at a screen wondering where to start now so I think you have achieved what you set out to do. Going through the services will be good. I’m looking forward to the one on functions as I don’t use them at the moment!

In regards to the IDs I am saying that I just assign the SVG element the ID which is one of my entities. So in your example I would have:
 ```
rules:
            - entity: light.hue1_light
              state_action:
                - action: call-service
                  service: floorplan.class_set
                  service_data: entitystate-${entity.state}
                - action: call-service
                  service: floorplan.style_set
                  service_data:
                    style: >
                      > var rgbColor = entity && entity.attributes &&
                      entity.attributes.rgb_color ? entity.attributes.rgb_color
                      : false; return rgbColor?
                      `--livingroom-rgb-color:rgb(${rgbColor});` : '';
```
and I would just assign an ID of `light.hue1_light` to the rectangle. Apart from removing a line in my yaml it also means I can search my SVG for the entity name to find the object I want to modify. The only time I see the need to use element is if two separate objects need to be controlled by the state of a single entity. What do you think?

**Reply by exetico** [reply]
😁😁 It's just OBS. 

No one else is spending their time creating the content. There's always room space to improvements no matter what, but I did take the previous feedback a little hard. That's why I replied as I did. I personally understod it more like '_Here is what I would have done cause the current solution is not that great. I could have done it better, for sure... but.... I dont really do that kind of things_'. 

But I've re-read the first feedback, and I'll think about everything next time 👍😁. 

I also get what you mean with the element-key. But It's important for the user to understand the connection, as the first thing.

**Reply by OzGav** [reply]
Sorry Tobias if my post didn't come across well on first reading. I am glad you can now see it for what it was. I am the first one to be grateful for all the support you are providing to the project.

I've just upgraded my computer to Windows 10 so I can run OBS! I hadn't heard of it before. I will try and make a video and teach myself how to use it at the same time. No promises on a timeline!

**Reply by exetico** [reply]
I'm not sure who to blame - and that's OK 😄. Could be the time of the day, the lack of coffee, or whatever else.

Video 3 has dropped - a video with focus on CSS, and more in-dept things. I've tried to think about most of the feedback from you. I've also spend a few hours on how to create some proper zoom and focus-animations.

🎥 👉 https://github.com/ExperienceLovelace/ha-floorplan/discussions/258

Now that english is not my first language, it's still hard to bring all facts on the table while speaking. I'm also doing this to force me into the usage of new words, but I'm not scripting anything. That's why there still a bit off room for improvements on the "endings" in each section 😄

**Reply by OzGav** [reply]
I though this one went well. Learning how to do the zoom and highlights was time well spent I think as it clearly targeted my attention to what you were trying to show and of course was easy to read even on my iPad. 👍

**Reply by exetico** [reply]
That sounds great. Thank you for the response. I found it a bit hard to select which parts required zoom or/and highlights, but i tried my best to not "abuse it", cause that would be bad, too. The good thing about uploading in 1080p is that I have a bit of "room" to zoom into the 1440p source. But I really need to separate my webcam and screen recordings - currently it's too time consuming to fixing the frame setup on every zoom (which is required with a webcam+screen recording in the same source-file). I'm not sure why I'm sharing it here... But now I've shared my most critical  next-up "work-flow" improvement 🎉...

<img width="1674" alt="image" src="https://user-images.githubusercontent.com/3549445/198066685-b7003712-97ca-4668-832f-02d8da63a51a.png">

_Visual example of "multi-source"_

**Comment by Lynx-eyed** [comment]
Sorry for extremely dumb questuion, I dont get YAML totally, but want to build floorplan... So I tried to follow the instructions in the video exactly, except that I did not find a _home-simple_ example in the repository. I tried _home_ instead, then also _floorplannet_home_, but I get the same error after inserting the code from the YAML file into the panel code: 'n: At path: views -- Expected an array value, but received: undefined'. I tried adding _views: []_, then there is no error, but the panel is also empty. What am I doing wrong?
![2023-01-22](https://user-images.githubusercontent.com/122067753/213910868-4dc6f6a8-9491-4a92-a2f2-91fc5d510c70.jpg)

**Reply by OzGav** [reply]
The attachments related to this video are here https://github.com/exetico/Home-Assistant_Attachments-for-YouTube/tree/main/1_Getting-Started-with-ha-floorplan

**Reply by Lynx-eyed** [reply]
thank you very much! I should have read the description of the video...

**Reply by exetico** [reply]
@Lynx-eyed Or, I should have pointed that out in the video. I'll try and remember that next time :smile: !

**Comment by W1ngR1d3r** [comment]
Thanks Tobias, great work!
I'd like to show a timer to the floorplan. I was able to add it and see the remaining time the following code:
                  state_action:
                    - service: floorplan.text_set
                      service_data: ${entity.attributes.remaining}
but it doesn't refresh (it refresh the time on state changes - start, pause, finish - but not every second).
Do you know how I can make it to refresh the timer every second?
![image](https://user-images.githubusercontent.com/107564886/217186751-806512a4-e7c6-4332-88c7-fd6da5340b0b.png)

**Reply by OzGav** [reply]
If you have an entity which has a constantly changing state you could use that as the entity (thus forcing a state_action often) and then access the timer attributes via the hass entity. This is the syntax to access the state of another entity so you would have to experiment to get the attributes.
`hass.states[”some.entity”].state`

edit: Actually you could try `- entity: '*'`

**Reply by W1ngR1d3r** [reply]
Thanks OzGav!
Sorry if I'm asking something obvious but I don't know how to do what you suggest.
This is the code I have and that is not refreshing the remaining time:

                - name: Luz Desayuno Timer
                  entities:
                    - timer.luz_desayuno_on_timer
                  state_action:
                    - service: floorplan.text_set
                      service_data: ${entity.attributes.remaining}                            
                    - service: floorplan.class_set
                      service_data:
                        class: static-temp
                    - service: floorplan.style_set
                      service_data:
                        element: timer.luz_desayuno_on_timer
                        style: |
                          >
                          if( entity.state === "active" )
                              return "fill: rgb(255, 255, 255, 0.7);";
                          return "fill: rgba(150, 150, 150, 1);";

Thanks in advance!

**Reply by OzGav** [reply]
I am on my iPad so can’t test this. The attributes entry might be wrong so you might have to experiment there. And I’m not 100% sure on that entity entry either but I saw it used in the examples. If that doesn’t work see if you have another entity which is constantly changing state.
```
            - name: Luz Desayuno Timer
              entity: “*”
              element: timer.luz_desayuno_on_timer
              state_action:
                - service: floorplan.text_set
                  service_data: ${hass.states["timer.luz_desayuno_on_timer”].attributes.remaining}                            
                - service: floorplan.class_set
                  service_data: static-temp
                - service: floorplan.style_set
                  service_data:
                    style: |
                      >
                      if( hass.states[”timer.luz_desayuno_on_timer”].state === "active" )
                          return "fill: rgb(255, 255, 255, 0.7);";
                      return "fill: rgba(150, 150, 150, 1);";
```

**Reply by W1ngR1d3r** [reply]
Thanks again OzGav!
I tried what you suggested but didn't work.
I will look for another entity that changes state constantly and give it a try.

**Reply by W1ngR1d3r** [reply]
I tried with a plug (W entity) and it works. It uses the entity state, so the challenge is to find how to make it work with an attribute.

**Reply by OzGav** [reply]
> I tried with a plug (W entity) and it works. It uses the entity state, so the challenge is to find how to make it work with an attribute.

Sorry I missed a quote in that line. I have fixed it above but there should be a " before timer. If you copy and paste it should work now

Also I wouldn't ordinarily have a `class_set` and` style_set` in the same rule. I suggest creating two classes that include those fill values and then just do a `class_set` based on the state (as you have for style). Alternatively if that `class_set` isn't doing anything (because you are setting the same class constantly regardless of state) then you could remove that and just keep the `style_set`

**Reply by exetico** [reply]
Thank you 😊. Let me know, if you need other inputs. It looks like OzGav is helping you here.

**Reply by W1ngR1d3r** [reply]
Tried again with the new code but still the same behavior. It refresh the text only when I pause the timer.

**Reply by OzGav** [reply]
Ok then put the plug entity in for now. 

@exetico any idea why that * entity wouldn’t work. It is working for me but I am using a beta as I am working with @pkozul

**Reply by exetico** [reply]
Hi @W1ngR1d3r (and @OzGav) . I'm sorry for the late reply. For me it looks like you're expecting ha-floorplan to return the remaining secounds in the svg, while it's counting down, right? I've not used the "Timer" integration before, but if I take a look, it looks like the timer is just a calculated field, on basis of the attributes of the timer. The "count" itself, doesn't really live within your Home Assistant engine. Instead it's handled on the client side.

Here's a bite of [state-card-timer.js](https://github.com/home-assistant/frontend/blob/dev/src/state-summary/state-card-timer.js):

https://github.com/home-assistant/frontend/blob/0b54e60b9169b63b2c3ffcc782401b15a858ac5e/src/state-summary/state-card-timer.js#L89

(check the link to see more)

That also tells us that the timer card in Home Assistant itself, uses the `timerTimeRemaining` from `../date/timer`. 

![image](https://user-images.githubusercontent.com/3549445/220298985-fa313d01-2019-4e00-b3ae-78c649ed85b2.png)

Therefore no entity-updates is done in the Home Assistant back-end nor front-end, and therefore, you're unable to use the entity by itself, to update the count-down.
_I must say that I'm 90% sure that a timer-example has been brought up before_, but maybe I should take that up as the next challenge for a video.

The above facts does explain why you have these kind of attributes, there the "remaining" attribute is controlling the time left in seconds.
![image](https://user-images.githubusercontent.com/3549445/220293466-87dd3b65-5d1e-423e-9525-15a7c13d8fc8.png)

I understand that this **doesn't** help you out with the needs, rather, it's hopefully help you understand what's normally going on behind the scenes in Home Assistant, in the Home Assistant source code. But if you're clever, it's totally possible to just use most of the HA source-code for this to work.

TL;DR: Timers is just rendered in the front-end. The "remaining" state is **not** updated, so you sadly can't expect ha-floorplan to count down for you :smile:. Instead, another solution needs to be implemented, acting like Home Assistant updates the time by itself, **or** simply creating everything by yourself :)

I'll get back with a solution :) 

In short, you'd need to "count by yourself, in a function", or, secure that the source-solution has the function build in (some custom-cards uses the "custom-card-helpers", but I'm not sure, if ha-floorplan should take this approach).

**Reply by exetico** [reply]
![image](https://user-images.githubusercontent.com/3549445/220402434-907bdf49-a9b0-416c-9a8d-d7571c7e5518.png)


Okay, solution is done, but needs a few refactoring things before I'll publish it. I also need an input from @pkozul, cause my plan is, to implement a event listener i ha-floorplan, so it's possible to call ha-floorplan services within a ha-floorplan execution (to use as ha-floorplan service like "variable_set").

Branch can be found here, but I'll not recommend to switch branch for this - wait for things to get into the main one, instead.

https://github.com/ExperienceLovelace/ha-floorplan/blob/73666328b8adb1f2fad7935cca4acba33a2bdae8/src/components/floorplan/floorplan-element.ts#L282

YAML for the current solution can be found here - but again, wait for the ha-floorplan release. I guess case4 will be used as a video, so a few rewrites of the JS and YAML will also be done, before that.
https://pastebin.com/G9eS4bAw
(Here I'm using variables, javascript, and the "event" solution, which are **not** in the main branch right now)

**Reply by W1ngR1d3r** [reply]
Amazing!!!
I will wait for the new release.
Thanks!!!!!

**Reply by exetico** [reply]
Hi, I'm still waiting on feedback, before I can release the feature - but hopefully it'll be ready in a short amount of time. I'm sorry about the delay :smile_cat:.

**Comment by Adz87kc** [comment]
Hi, @exetico 

I've been looking at this for a few hours this afternoon and I still can't get it to work on my HA.
The only part that didn't go as per the YouTube video was at 6:00 where you confirm floorplan.js is installed.

Instead, I clicked on: add resource -> JavaScript Module and typed in: /hacsfiles/ha-floorplan/floorplan.js


I did everything else as the video instructed. Is there anything else I can look at? 




I installed via HACS, uploaded the css and SVG files using file editor into the folder described in the video. I even used the same file names as the yaml code provided incase I had a typo.

**Reply by exetico** [reply]
Hi :-) Can you find the HACS files in your config folder? Have the system been restarted here after? Have you tried `www` instead of `hacsfiles`?

What kind of errors can you see in the browser console, if you open up the browser inspect tool? :-)

I hope some of the questions can help you with the next step(s). Else, fell free to come back with more details.

**Reply by Adz87kc** [reply]
Nvm, I restarted HA and it appeared!
Now onto the rest of the video :D


Edit: additional HA restart _after_ adding the Yaml code.

**Reply by exetico** [reply]
That's actually mentioned in the video. But that's great to hear! Happy floorplanning 😁!

**Comment by landon-hillyard** [comment]
One more thing, related to this post video, but a slightly different challenge. I made an SVG and copied some of the objects you created in your SVG file. I then uploaded my edited SVG to start working on my layout. I have attached the SVG that I have uploaded as well a screen shot of what I am seeing. 

Any idea on how to get floorplan to reference my new SVG vs the original? 

Is there some sort of cache I need to clear or something? 


![home_simple](https://github.com/ExperienceLovelace/ha-floorplan/assets/43597100/fa04dafc-2978-4148-b932-4d098db31f29)



![Capture](https://github.com/ExperienceLovelace/ha-floorplan/assets/43597100/2fce9912-1716-4dcc-a14b-c78bfbb04e5e)

**Reply by exetico** [reply]
Hi. Yes, there is. But there's multiple ways to handle this. I'm pretty sure I've already mentioned this in on of the videos, but I'm not sure, in which of them its stuffed into.

In your case, you can just upload your new version, and change the filename.

Personally I open the full URL to my SVG file in my browser, and force reload the page (Ctrl+Shift+R). That normally works, if the caching are playing with me.

**Reply by landon-hillyard** [reply]
Ok, I think this makes sense. I have made a video to ensure the issue I am struggling with is clear. I try what you suggested and report back with the results. 
https://www.youtube.com/watch?v=vW7exyDLVZA&t=2s

**Comment by sergentum** [comment]
Can't make it work. Resources contain js, but still it doesn't work. 
Another js auto-entities works, btw.
![2023-07-14 23_19_52](https://github.com/ExperienceLovelace/ha-floorplan/assets/4054762/e65ca51d-7162-4e6b-9616-9c1704aec4bd)
![2023-07-14 23_18_34](https://github.com/ExperienceLovelace/ha-floorplan/assets/4054762/4ca89667-416f-4fc8-ab67-53dbe56bacb6)

**Reply by OzGav** [reply]
Can you show the screen capture where the resource is loaded

**Reply by sergentum** [reply]
I fixed the problem by updating HA os and core. I had 2023.4 and current version 2023.7 so now it started to work.

**Comment by agouveia19** [comment]
I followed everything in your video, but I get this error.. I can't for the life of me figure out what's wrong with it.. Can someone please help?

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/142179234/c1687d4e-0c6d-4dbc-b624-37524471de9e)

**Reply by OzGav** [reply]
`home_simple.css` is either missing, not at that path or there is a permission problem. Type `http://<YOUR_HA_IP>:8123/local/floorplan/home_simple/home_simple.css`into your browser address bar and see if the contents of the file appear

**Reply by agouveia19** [reply]
@OzGav I don't know what happened, but when I just started HA it is working as the video. Thank you anyway!😊

**Reply by exetico** [reply]
Maybe you forgot to restart HA, as stated in the video? 😄 But it's great that it works now 👏 !

**Comment by mesterila** [comment]
I have a small problem with loading the SVG file. Everything was done to the letter (with my own files) - and it failed to load.

Backtracing, I downloaded the example files, and uploaded/used them (including the config) - also failed to load the home.svg image. 
The CSS is not empty (I used all files and config as in the example).
Checking this article - I don't see anything missing: https://www.home-assistant.io/dashboards/picture-elements/#how-to-use-the-style-object
The path to the files is correct (I use the full path /config/www/community/hafloorplan/ - and changing it produces a clear error). Notice that I removed the "-" and the extra directories to eliminate any typos or problems related to paths.

There is clearly a difference between "Raw configuration editor" > saves (no error)> loads a blank page, and "Edit in YAML" which displays a big blue logo of HA (that is not the .SVG file I try to load, nor the home.svg used in the example). I saw this behavior in another post - with screenshots, but I can't find that. Workaround was to restart the HA. I did, but without luck. 

Note: Running latest HAOS 2024.2.1, on freshly deployed HA Green

The only errors I receive in console are related to the HA entities that don't exist (I didn't change any entity from the example yet).
I did this at least 15 times, cleaning everything before a fresh start - and I still can't figure out what's wrong. 

Any hints? Something I should check before I go crazy?

**Reply by mesterila** [reply]
These are the only console errors I receive (which I think they are normal):

2/16/2024, 6:06:25 PM INFO INIT Floorplan for Home Assistant (ha-floorplan) v1.0.43
3
floorplan.js?hacstag=1883234941043:54 2/16/2024, 6:06:29 PM WARNING CONFIG Cannot find 'light.garage' in Home Assistant entities
3
floorplan.js?hacstag=1883234941043:54 2/16/2024, 6:06:29 PM WARNING CONFIG Cannot find 'switch.living_area_fan' in Home Assistant entities
floorplan.js?hacstag=1883234941043:54 2/16/2024, 6:06:29 PM WARNING CONFIG Cannot find 'camera.zagreb_ban_jelacic_square' in Home Assistant entities
floorplan.js?hacstag=1883234941043:54 2/16/2024, 6:06:29 PM WARNING CONFIG Cannot find 'binary_sensor.main_bedroom' in Home Assistant entities
floorplan.js?hacstag=1883234941043:54 2/16/2024, 6:06:29 PM WARNING CONFIG Cannot find 'binary_sensor.living_area' in Home Assistant entities
0:1 Unchecked runtime.lastError: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
0:1 Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received

But the screen is stuck at this even after restart ( I also reinstalled the HACS and the floorplan-ha in the meantime):
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/355a8134-b797-43be-ae3d-46ab9a5bc722)

**Reply by OzGav** [reply]
Start by putting the path to your SVG in your browser address bar and confirm that it loads correctly.

**Reply by mesterila** [reply]
Thanks for checking 
But it doesn't load (anything). If I load in broser the HA including the port, it just loads a blank page in the Overview tab (for any file I tested, even a random JPG I placed there):
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/6a53ddad-1e65-47b0-9447-29e10aa62f3f)
But the files are there if I access via smb:
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/afe68bcc-ac84-4b72-8b2f-e4dcc30105ca)

**Reply by mesterila** [reply]
and to have the complete data, here is the YAML I used for this:
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/bd843a9b-89f9-4132-a101-03c79673fe53)

**Reply by mesterila** [reply]
And this is trying to load the image in browser:
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/bbb59dc8-7ef9-4e8b-ac89-07414bc742b1)
![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/59647895/28cf90ae-908a-4b8f-82e8-79cf72e2948a)

**Reply by OzGav** [reply]
Try `/local/community/ha-floorplan/home.svg`

**Reply by exetico** [reply]
There's something wrong with your path. The first part would normally start with `/local` or `/www`. Never `/config`. Your start-up problem is related to your Home Assistant setup, and not related to ha-floorplan. 

Try a few more paths, also as simple as possible. Also, I'd never recommend to use our solutions folder for your userfiles. HACS could throw that away at some point, or maybe run into problems. Make your own folder in the Home Assistant > `www` folder. Normally `www` are resolved as `local` in your webbrowser.

The screenshot your providing of your files does not translate directly to the HTTP resource path, as clearly stated in the video, too 😊.

Happy troubleshooting!

**Reply by mesterila** [reply]
Thank you @OzGav  - that did the trick. I swear I tried this before - probalby the very first thing, otherwise I wouldn't change it to something else
@exetico - thanks for the explanation ;) I'm learning, few hours every day it's quite a challenge
